### PR TITLE
npm: enable debug feature parity for v2

### PR DESCRIPTION
**Pull request by matmerr**
**Reason for Change**:

Enables basic debug feature parity with V1 and V2. Followup PR will add bubble up iptables comments along with gettuples.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

**Comment by rbtr** [comment]
@vakalapa you might want to update the [reviewers team](https://github.com/orgs/Azure/teams/acn-npm-reviewers) 🙂

File: npm/util/ioutil/restore.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func (creator *FileCreator) RunCommandWithFile(cmd string, args ...string) error
 		if wasFileAltered {
 			sameNew = "updated"
 		}
-		msg := fmt.Sprintf("on try number %d, failed to run command [%s]. Rerunning with %s file. Had error [%s].\nUsed file:\n%s", creator.tryCount, commandString, sameNew, err.Error(), fileString)
+		msg := fmt.Sprintf("on try number %d, failed to run command [%s]. Rerunning with %s file. Had error [%s].Used file:%s", creator.tryCount, commandString, sameNew, err.Error(), fileString)
```
[the remark this answers is not in the document]
Comment (rbtr) [reply npm/util/ioutil/restore.go:144]: logs shouldn't be multiline. one log per line pls

File: npm/cmd/gettuples.go [hunk 2]
```diff
@@ -25,18 +29,34 @@ func newGetTuples() *cobra.Command {
...
-				_, tuples, err := dataplane.GetNetworkTuple(srcInput, dstInput)
+				config := &npmconfig.Config{}
+				err := viper.Unmarshal(config)
 				if err != nil {
-					return fmt.Errorf("%w", err)
+					log.Printf("failed to load config with err ")
 				}
-				for _, tuple := range tuples {
-					fmt.Printf("%+v\n", tuple)
+				if config.Toggles.EnableV2NPM {
```
Comment (matmerr) [inline npm/cmd/gettuples.go:42]: redundant block as below

File: npm/pkg/dataplane/debug/trafficanalyzer.go [hunk 3]
```diff
@@ -414,6 +383,8 @@ func matchCIDRBLOCKS(pod *controllersv1.NpmPod, setInfo *pb.RuleResponse_SetInfo
 			}
 		}
 	}
+
+	log.Printf("matched cidr")
```
Comment (matmerr) [inline npm/pkg/dataplane/debug/trafficanalyzer.go:387]: remove

File: npm/pkg/dataplane/debug/trafficanalyzerv2.go [hunk 4]
```diff
@@ -0,0 +1 @@
+package debug
```
Comment (matmerr) [inline npm/pkg/dataplane/debug/trafficanalyzerv2.go:1]: remove

File: npm/pkg/dataplane/ipsets/ipsetmanager.go [hunk 5]
```diff
@@ -446,16 +446,14 @@ func (iMgr *IPSetManager) ApplyIPSets() error {
...
 
-func (iMgr *IPSetManager) GetAllIPSets() []string {
-	iMgr.Lock()
-	defer iMgr.Unlock()
-	setNames := make([]string, len(iMgr.setMap))
-	i := 0
-	for setName := range iMgr.setMap {
-		setNames[i] = setName
-		i++
+func (iMgr *IPSetManager) GetAllIPSets() map[string]string {
```
Comment (matmerr) [inline npm/pkg/dataplane/ipsets/ipsetmanager.go:449]: add comment what return map key and value are

File: npm/npm.go [hunk 6]
```diff
@@ -90,62 +94,38 @@ func NewNetworkPolicyManager(config npmconfig.Config,
...
-		return nil, errors.Errorf("%s: %v", models.ErrMarshalNPMCache, err)
-	}
-	m[models.NsMap] = npmNamespaceCacheRaw
+	var cacheRaw []byte
 
-	var podControllerRaw []byte
 	if npMgr.config.Toggles.EnableV2NPM {
-		podControllerRaw, err = json.Marshal(npMgr.PodControllerV2)
+		cachev2 := controllersv2.Cache{}
+		cachev2.NsMap = npMgr.NamespaceControllerV2.GetCache()
```
Comment (matmerr) [inline npm/npm.go:105]: condense into struct declaration

File: npm/pkg/dataplane/debug/trafficanalyzer.go [hunk 7]
```diff
@@ -389,13 +356,15 @@ func matchNAMEDPORTS(pod *controllersv1.NpmPod, setInfo *pb.RuleResponse_SetInfo
 			} else {
 				rule.DPort = namedPort.ContainerPort
 			}
+
+			log.Printf("matched named ports")
```
Comment (matmerr) [inline npm/pkg/dataplane/debug/trafficanalyzer.go:360]: remove

File: npm/pkg/dataplane/debug/trafficanalyzer.go [hunk 8]
```diff
@@ -368,10 +334,11 @@ func matchKEYLABELOFPOD(pod *controllersv1.NpmPod, setInfo *pb.RuleResponse_SetI
 		// if key does not exist but required in rule
 		return false
 	}
+	log.Printf("matched key label of pod")
```
Comment (matmerr) [inline npm/pkg/dataplane/debug/trafficanalyzer.go:337]: remove

File: npm/pkg/dataplane/debug/trafficanalyzer.go [hunk 9]
```diff
@@ -327,39 +288,44 @@ func matchNESTEDLABELOFPOD(pod *controllersv1.NpmPod, setInfo *pb.RuleResponse_S
...
 	return true
 }
 
-func matchKEYVALUELABELOFPOD(pod *controllersv1.NpmPod, setInfo *pb.RuleResponse_SetInfo) bool {
+func matchKEYVALUELABELOFPOD(pod *common.NpmPod, setInfo *pb.RuleResponse_SetInfo) bool {
 	key, value := processKeyValueLabelOfPod(setInfo.Name)
 	if pod.Labels[key] != value || (pod.Labels[key] == value && !setInfo.Included) {
 		return false
 	}
+	log.Printf("matched key value label of pod")
```
Comment (matmerr) [inline npm/pkg/dataplane/debug/trafficanalyzer.go:324]: remove

File: npm/http/api/api.go [hunk 10]
```diff
@@ -6,6 +6,7 @@ const (
 	NodeMetricsPath    = "/node-metrics"
 	ClusterMetricsPath = "/cluster-metrics"
 	NPMMgrPath         = "/npm/v1/debug/manager"
+	NPMV2CacheAPI      = "/npm/v2/debug/cache"
```
Comment (vakalapa) [inline npm/http/api/api.go:9]: Not needed, since marshall takes care of this now.

File: npm/pkg/dataplane/debug/trafficanalyzer.go [hunk 11]
```diff
@@ -129,38 +106,19 @@ func getNetworkTupleCommon(
...
-	case EXTERNAL:
-		return &controllersv1.NpmPod{}, nil
-	default:
-		return nil, errInvalidInput
-	}
-}
-
 // GetInputType returns the type of the input for GetNetworkTuple.
-func GetInputType(input string) InputType {
+func GetInputType(input string) common.InputType {
```
Comment (vakalapa) [inline npm/pkg/dataplane/debug/trafficanalyzer.go:110]: move this function to where input type is defined.

File: npm/pkg/dataplane/debug/trafficanalyzer.go [hunk 12]
```diff
@@ -206,46 +164,47 @@ func generateTuple(src, dst *controllersv1.NpmPod, rule *pb.RuleResponse) *Tuple
...
+
 	for _, rule := range rules {
-		matched := true
+		matched := false
+		// evalute all match set in src
 		for _, setInfo := range rule.SrcList {
-			// evalute all match set in src
 			if src.Namespace == "" {
 				// internet
 				matched = false
```
Comment (vakalapa) [inline npm/pkg/dataplane/debug/trafficanalyzer.go:180]: redundant remove line

File: npm/pkg/dataplane/debug/trafficanalyzer.go [hunk 13]
```diff
@@ -206,46 +164,47 @@ func generateTuple(src, dst *controllersv1.NpmPod, rule *pb.RuleResponse) *Tuple
...
 	rules []*pb.RuleResponse,
-	npmCache *controllersv1.Cache,
+	npmCache common.Cache,
 ) ([]*pb.RuleResponse, error) {
 
 	res := make([]*pb.RuleResponse, 0)
+
 	for _, rule := range rules {
-		matched := true
+		matched := false
```
Comment (vakalapa) [inline npm/pkg/dataplane/debug/trafficanalyzer.go:175]: In V2 we have cut down a single rule, into a half match in top level ingress or egress chain and then jump to Netpol specific ingress or egress chain and match the 2nd half of the packet in a given direction, that change is still missing

**Review by vakalapa — CHANGES_REQUESTED** [review]

**Review by huntergregory — COMMENTED** [review]
converter logic looks good now

**Pull request merged**